imap_processing/tests/idex/conftest.py

```diff
@@ -48,3 +69,96 @@ def get_spice_data_side_effect_func(l1a_ds, idex_attrs):
         )
         for name in SPICE_ARRAYS
     }
+
+
+def load_hdf_file(path: str, num_events: Optional[int] = None) -> dict:
+    """
+    Loads an HDF5 file produced by the IDEX team into a dictionary where the keys
+    correspond to the groups in the file, and the values are arrays containing the data.
+
+    The HDF5 files are organized by event number, where each event contains groups
+    for metadata and data items. This function consolidates all events into a single
+    array to be consistent with the cdf organization.
+    For example, if there are 14 events and the metadata value "category" is extracted,
+    it will appear in the output dictionary as:
+    {"category": np.ndarray of shape (14,)}.
+
+    Parameters:
+    ----------
+    path : str
+        The file path to the HDF5 file.
+    num_events : int, optional
+        The number of events to extract. If None, all events in the file are processed.
+
+    Returns:
+    -------
+    dict
+        A dictionary containing the extracted data.
+
+    Warnings:
+    --------
+    - This function assumes the HDF5 file is structured with top-level groups named
+      numerically (e.g., "1", "2", "3") representing events.
+    """
+
+    def collect_arrays(name, obs):
+        """
+        Collects data from HDF5 datasets into the dictionary `data_vars`.
+
+        Parameters:
+        ----------
+        name : str
+            The full path of the dataset in the HDF5 file.
+        obs : h5py.Dataset or h5py.Group
+            The object at the current position in the HDF5 file.
+        """
+        # Split the full dataset path to extract the group key and event number
+        names = name.split("/")
+        group_key = names[-1]
+        event = names[0]
```

Comment:
```suggestion
event, group_key = name.split("/")
```
I think you could do this? Just a tad more concise.